src/Serialization/GAP.jl

```diff
@@ -117,31 +121,42 @@ install_GAP_serialization(:IsFreeGroup,
     end
   end)
 
-install_GAP_deserialization(:IsFreeGroup,
-  function(filt::GapObj, s::DeserializerState, T, d::Dict)
-    if haskey(d, :freeGroup) && haskey(d, :gens)
-      # Deserialize the full free group.
-      F = load_typed_object(s, d[:freeGroup])
-      # Deserialize the generators.
-      generators = load_object(s, Vector, d[:gens], (Vector{Int}, Int))
-      fam = GAPWrap.ElementsFamily(GAPWrap.FamilyObj(F))
-      Ggens = [GAPWrap.ObjByExtRep(fam, GapObj(x, true)) for x in generators]
-      # Create the subgroup.
-      G = GAP.Globals.SubgroupNC(F, GapObj(Ggens))
-    else
-      # Create a new full free group.
-      wfilt = getproperty(GAP.Globals, Symbol(d[:wfilt]))
-      if haskey(d, :nameprefix)
-        # infinite rank
-        prefix = GapObj(d[:nameprefix])
-        init = GapObj(d[:names], true)
-        G = GAP.Globals.FreeGroup(wfilt, GAP.Globals.infinity, prefix, init)
+install_GAP_deserialization(
+  :IsFreeGroup,
+  function(filt::GapObj, s::DeserializerState, T)
+    load_node(s) do d
+      if :freeGroup in keys(d) && :gens in keys(d)
+        # Deserialize the full free group.
+        F = load_typed_object(s, :freeGroup)
+        # Deserialize the generators.
+        generators = load_object(s, Vector, (Vector{Int}, Int), :gens)
+        fam = GAPWrap.ElementsFamily(GAPWrap.FamilyObj(F))
+        Ggens = [GAPWrap.ObjByExtRep(fam, GapObj(x, true)) for x in generators]
+        # Create the subgroup.
+        G = GAP.Globals.SubgroupNC(F, GapObj(Ggens))
       else
-        names = GapObj(d[:names], true)
-        G = GAP.Globals.FreeGroup(wfilt, names)
+        # Create a new full free group.
+        wfilt = getproperty(GAP.Globals, load_object(s, Symbol, :wfilt))
+        if :nameprefix keys(d)
```

Comment:
An `in` is missing here.
(I had thought that `haskey(d, :nameprefix)` is better, since one need not create `keys(d)`.)

Reply:
d is now a JSON3.Object longer and no longer has the haskey method. Thanks for spotting that, weird that I didn't get a revise error